src/select-query-parser/parser.ts

```diff
@@ -220,13 +221,24 @@ type ParseNonEmbeddedResourceField<Input extends string> = ParseIdentifier<Input
 ]
   ? // Parse optional JSON path.
     (
-      Remainder extends `->${infer _}`
+      Remainder extends `->${infer PathAndRest}`
         ? ParseJsonAccessor<Remainder> extends [
             infer PropertyName,
             infer PropertyType,
             `${infer Remainder}`
           ]
-          ? [{ type: 'field'; name: Name; alias: PropertyName; castType: PropertyType }, Remainder]
+          ? [
+              {
+                type: 'field'
+                name: Name
+                alias: PropertyName
+                castType: PropertyType
+                jsonPath: JsonPathToAccessor<
```

Comment:
We use this to format and save the value "path accessor" information from the parser to the result inference. That way, the parser can still be agnostic to the database types, only caring about string parsing.